### PR TITLE
Create metadata API endpoint

### PR DESCRIPTION
**Pull request by basiliskus**
# Create metadata API endpoint

- Created `/v1/etor/metadata` endpoint
- Added `pathParams` to `DomainRequest` in order to get path parameters when using GET
- Added openapi docs and tests

## Issue

#672 

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 1]
```diff
@@ -131,4 +138,16 @@ DomainResponse handleOrders(DomainRequest request) {
...
+
+    DomainResponse handleMetadata(DomainRequest request) {
+        try {
+            String uniqueId = request.getBody();
+            PartnerMetadata metadata = partnerMetadataStorage.readMetadata(uniqueId);
+            return domainResponseHelper.constructOkResponse(
+                    formatter.convertToJsonString(metadata));
+        } catch (PartnerMetadataException | FormatterProcessingException e) {
+            logger.logError("Unable to read metadata", e);
+            return domainResponseHelper.constructErrorResponse(400, e);
```
Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:150]: This should be a `500`, you think?  Both those exceptions are generated when bad stuff happens and there's nothing a caller can do about it, right?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 2]
```diff
@@ -131,4 +138,16 @@ DomainResponse handleOrders(DomainRequest request) {
         OrderResponse orderResponse = new OrderResponse(orders);
         return domainResponseHelper.constructOkResponse(orderResponse);
     }
+
+    DomainResponse handleMetadata(DomainRequest request) {
+        try {
+            String uniqueId = request.getBody();
```
Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:144]: I think it makes more sense to get the unique ID from the URL path.  E.g. `/v1/etor/metadata/{uniqueId}`.

**Review by halprin — COMMENTED** [review]
A couple of recommendations.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:150]: Right! Good call

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 2 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:144]: Agreed. I'll change that

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java]: Fixed

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java (hunk 2 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java]: Done

File: etor/src/main/resources/openapi_etor.yaml [hunk 3]
```diff
@@ -74,6 +74,43 @@ paths:
...
+          content:
+            application/json:
+            schema:
+              $ref: '#/components/schemas/NotFoundError'
+        '500':
+          description: Unable to retrieve requested metadata
+          content:
+            application/json:
+            schema:
+              $ref: '#/components/schemas/InternalServerError'
```
Comment (halprin) [inline etor/src/main/resources/openapi_etor.yaml:113]: I wouldn't include `500`.  That's when something goes big boom and generally isn't expected given any input from a user.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java [hunk 4]
```diff
@@ -131,4 +140,23 @@ DomainResponse handleOrders(DomainRequest request) {
         OrderResponse orderResponse = new OrderResponse(orders);
         return domainResponseHelper.constructOkResponse(orderResponse);
     }
+
+    DomainResponse handleMetadata(DomainRequest request) {
+        try {
+            String metadataId = request.getPathParams().get("id");
```
Comment (jcrichlake) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:146]: Do we know if this is the submission id at this point? Or is this just an ID that will be determined later?

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java:146]: It's the id RS is working on sending to us, which I think it'd be the original submissionId

**Review by tjohnson7021 — APPROVED** [review]

**Review by halprin — APPROVED** [review]

**Review by jcrichlake — APPROVED** [review]

**Review by jorg3lopez — APPROVED** [review]
Great job, looks good!

**Pull request merged**